discovery-provider/src/queries/queries.py

```diff
@@ -188,8 +186,17 @@ def get_tracks_including_unlisted():
                 )
 
         # Perform the query
+        # TODO: pagination is broken with unlisted tracks
         query_results = paginate_query(base_query).all()
         tracks = helpers.query_result_to_list(query_results)
+
+        # Mapping of track_id -> track; used to check route_id when iterating through identifiers
+        track_map = {t["track_id"]: t for t in tracks}
+
+        # If the track is unlisted and the generated route_id does not match the route_id in db,
+        # filter track out from response
+        tracks = [track_map[i["id"]] for i in identifiers if not (track_map[i["id"]]["is_unlisted"] and \
+                    track_map[i["id"]]["route_id"] != helpers.create_track_route_id(i["url_title"], i["handle"]))]
```

Comment:
this logically looks good, but this line is pretty hard to read, so would be good to separate it out. something like this:

```
# Perform the query
query_results = paginate_query(base_query).all()
tracks = helpers.query_result_to_list(query_results)

identifiers_map = {track["id"]: track for track in identifiers}

def filter_fn(track):
    input_track = identifiers_map[track["track_id"]].
    route_id = helpers.create_track_route_id(input_track["url_title"], input_track["handle"])
    return (!track["is_unlisted"] or track["route_id"] == route_id)

tracks = list(filter(filter_fn, tracks))

track_ids = list(map(lambda track: track["track_id"], tracks))
```